BasicSample/Assets/AppRemoting/Scripts/AppRemotingSample.cs

```diff
@@ -84,25 +103,37 @@ private void Update()
                             break;
                         case Remoting.ConnectionState.Disconnected:
                             ShowConnection2DUI();
-                            DisconnectFromRemote();
                             break;
                     }
                 }
             }
```

Comment:
here need a else condition.  otherwise the message below will get stuck at "connecting..." if user input wrong ip address

```
            else
            {
                m_connectionState = Remoting.ConnectionState.Disconnected;
            }
```

Reply:
this condition represent there's no XR session currently running.  you can use this to differentiate the "listening vs. disconnected" at line 123 and 124 below.

Reply:
Here is the tricky part with combining messages for Connect and Listen modes. What you suggested above works very well with connect. But in Listen mode, when I set the m_connectionState to Disconnected when connection state is invalid, it displays "Disconnected" and never displays the "Listening to incoming connection on ip" message. Probably, I can get away with adding the else condition as suggested above which helps the connect mode and checking for connectionStateValid for listen mode as below.